elements/lisk-p2p/src/peer_directory/new_peers.ts

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { P2PPeerInfo } from '../p2p_types';
+import { constructPeerIdFromPeerInfo, getBucket, PEER_TYPE } from '../utils';
+
+export const DEFAULT_NEW_PEER_BUCKET_COUNT = 128;
+export const DEFAULT_NEW_PEER_BUCKET_SIZE = 32;
+export const DEFAULT_EVICTION_THRESHOLD_TIME = 86400000; // Milliseconds in a day -> hours*minutes*seconds*milliseconds;
+
+export interface NewPeerConfig {
+	readonly evictionThresholdTime?: number;
+	readonly newPeerBucketCount?: number;
+	readonly newPeerBucketSize?: number;
+	readonly secret: number;
+}
+interface NewPeerInfo {
+	readonly peerInfo: P2PPeerInfo;
+	readonly dateAdded: Date;
+}
+
+export interface AddPeerOutcome {
+	readonly success: boolean;
+	readonly isEvicted: boolean;
+	readonly evictedPeer?: P2PPeerInfo;
+}
+export class NewPeers {
```

Comment:
`triedPeers` and `newPeers` classes are 90% the same. I suggest to create a `base_list` like we did with the peers and extend it. There will be less duplicated code and it will be easy to understand what are the differences in between the two current lists. Besides, we will allow for creating more lists in the `peer_directory` as it can be `bannedPeers`.

Reply:
I agree, it's better if they share/reuse code.
It could be done as a separate issue though.

Reply:
Issue created https://github.com/LiskHQ/lisk-sdk/issues/4048